pandas/core/internals/blocks.py

```diff
@@ -576,7 +576,10 @@ def convert_dtypes(
     @final
     @cache_readonly
     def dtype(self) -> DtypeObj:
-        return self.values.dtype
+        try:
+            return self.values.dtype
+        except AttributeError:  # PyArrow fallback
+            return self.values.type
```

Comment:
This doesn't make sense to me.  self.values should be the EA, and the EA.dtype should be the right thing here.

Reply:
Ah OK thanks. I think this is a holdover from an intermediate state and I didn't recognize the requirement here. Reverting this fixes a lot of the other comments you've made here as well - thanks!